superset/connectors/sqla/models.py

```diff
@@ -261,9 +261,9 @@ class SqlaTable(Model, BaseDatasource):
     owner_class = security_manager.user_model
 
     __tablename__ = 'tables'
-    __table_args__ = (UniqueConstraint('database_id', 'table_name'),)
+    __table_args__ = (UniqueConstraint('database_id', 'schema', 'table_name'),)
 
-    table_name = Column(String(250))
+    table_name = Column(String(127), nullable=False)
```

Comment:
Mmmh, are you sure you all don't have a table with 127+ chars?

Reply:
@mistercrunch some SQL Lab tables (views) are more than 127 characters. We can definitely make a change to reduce this for future views. I'm not certain what other viable options we have if we want to make sure that the full table name is unique.

Reply:
I feel like we may have been going back and forth on this in the past, but if you check and raise in the corresponding ModelView's `pre_add` method, it should do the right thing for the corresponding REST endpoint and CRUD auto form in FAB. 

It won't do it if/where the code uses the SQLAlchemy ORM directly (if any there are any occurrences) of that in the codebase

Reply:
Oh look at this:
https://github.com/apache/incubator-superset/blob/master/superset/connectors/sqla/views.py#L296-L303

Reply:
@mistercrunch what about if some renames a datasource using the React table editor?

Reply:
Oh right, we should refactor and run this and execute it on pre-update too, and yes squeeze in the logic on the endpoint that `<TableEditor/>` uses...

Reply:
@mistercrunch though I agree this _should_ work it seems like the correct solution would be to enforce this uniqueness at the data model level (as opposed to UI validation). I’ll explore whether there are other solutions to MySQL’s 767 byte limitation.

Reply:
@mistercrunch [here's](https://stackoverflow.com/a/35851332) a suggestion for increasing the maximum column size. The issue is this requires an engine configuration change which I assume is non-ideal.

Reply:
Mmmh, how about putting a unique constraint on a large 128 SHA based on the concat of the fields?